Utils.py

```diff
@@ -1074,3 +1075,86 @@ def is_iterable_except_str(obj: object) -> TypeGuard[typing.Iterable[typing.Any]
     if isinstance(obj, str):
         return False
     return isinstance(obj, typing.Iterable)
+
+
+def build_sphinx_docs() -> None:
+    """Build Sphinx autodocs."""
+    # noinspection PyUnresolvedReferences
+    from sphinx.cmd.build import main as sphinx_main
+
+    base_dir = os.path.dirname(__file__)
+    docs_path = os.path.join(base_dir, "docs")
+    sphinx_input = os.path.join(docs_path, "sphinx", "source")
+    sphinx_output = os.path.join(base_dir, "build")
+
+    # copy markdown files to sphinx directory to get rendered
+    for file in os.scandir(docs_path):
+        if file.name.endswith(".md"):
+            shutil.copy(file, sphinx_input)
+            # parse through the file and fix links for sphinx's api
+            with open(os.path.join(sphinx_input, file.name), "r") as f:
+                lines = f.readlines()
+            # starting at -1 so i can iterate it early instead of in every if block
+            line_index = -1
+            while line_index < len(lines):
+                line_index += 1
+                line = lines[line_index]
```

Comment:
this always crashes every list as-is lmao
```suggestion
            while line_index + 1 < len(lines):
                line_index += 1
                line = lines[line_index]
```